src/main/drivers/pwm_output_dshot_hal.c

```diff
@@ -238,6 +238,25 @@ bool pwmDshotMotorHardwareConfig(const timerHardware_t *timerHardware, uint8_t m
         return false;
     }
 
+    dmaIdentifier_e dmaIdentifier = dmaGetIdentifier(dmaRef);
+
+    bool dmaIsConfigured = false;
+#ifdef USE_DSHOT_DMAR
+    if (useBurstDshot) {
+        const resourceOwner_t *owner = dmaGetOwner(dmaIdentifier);
+        if (owner->owner == OWNER_TIMUP && owner->resourceIndex == timerGetTIMNumber(timerHardware->tim)) {
+            dmaIsConfigured = true;
+        } else if (!dmaAllocate(dmaIdentifier, OWNER_TIMUP, timerGetTIMNumber(timerHardware->tim))) {
+            return false;
+        }
+    } else
+#endif
+    {
+        if (!dmaAllocate(dmaIdentifier, OWNER_MOTOR, RESOURCE_INDEX(reorderedMotorIndex))) {
```

Comment:
In `resource.h` we have `#define RESOURCE_INDEX(x) (x + 1)` but in `cli.c` we have `#define MAX_RESOURCE_INDEX(x) ((x) == 0 ? 1 : (x))`. If a resource index means the same in both contexts thus could lead to confusion. Not really directly related to this PR; just an observation.

Reply:
Good find - we might be able to simplify.

Reply:
Na, I checked. `MAX_RESOURCE_INDEX()` is used to interpret `maxIndex` in `cliResourceValue_t`, and deal with indexed vs. non-indexed resources. But it could probably do with a better name.

Reply:
See #10969.